actors/util/math/exp.go

```diff
@@ -0,0 +1,29 @@
+package math
+
+import "github.com/filecoin-project/go-state-types/big"
+
+// ExpBySquaring takes a Q.128 base b and an int64 exponent n and computes n^b
+// using the exponentiation by squaring method, returning a Q.128 value.
+func ExpBySquaring(base big.Int, n int64) big.Int {
+	one := big.Lsh(big.NewInt(1), Precision128)
+	// Base cases
+	if n == 0 {
+		return one
+	}
+	if n == 1 {
+		return base
+	}
+
+	// Recurse
+	if n < 0 {
+		inverseBase := big.Div(big.Lsh(one, Precision128), base) // Q.256 / Q.128 => Q.128
+		return ExpBySquaring(inverseBase, n)
```

Comment:
This should be `-n` to prevent infinite regress, since `n` is already negative.

It doesn't look like we ever exercise this, so adding an error return and refusing negative exponents would also be ok.